File: faculdade/ProjectosAC/java/AC2122-ficha01/Exercicio5.java
import java.util.*;



public class Main{

	public static void main(String[] args){
	
	System.out.println("Insere o numero a testar");
	Scanner in= new Scanner(System.in);
	
	
	}
private void binoper(int x) {

  System.out.printf("\nb = %d %x\n", x, x);
  
  int resultado = x; // completar
  
  //intersecao de 1 com o numero a avaliar. Sera 0 se for par e 0 se for impar
  int aux=Integer.toBinaryString(2);
  resultado = x&aux==1;
  
  System.out.printf("a) %d  %x\n", resultado,resultado );
  
  /*Resultado dividido por 2^2=4*/
  resultado= x>>2;
  
  System.out.printf("b) %d  %x\n", resultado, resultado);
  
  resultado= (x<<3)+(x<<2);
  System.out.printf("c) %d  %x\n", resultado, resultado);
}
}

File: faculdade/ProjectosAC/C/sources/exercicio5.c

#include <stdio.h>

void binoper(int x) {

  printf("\nb = %d %x\n", x, x);
  int resultado = x; // completar
  
  //intersecao de 1 com o numero a avaliar. Sera 0 se for par e 0 se for impar
  int aux=0b0000000000000000000000000001;
  resultado = x&aux==1;
  
  printf("a) %d  %x\n", resultado,resultado );
  
  /*Resultado dividido por 2^2=4*/
  resultado= x>>2;
  
  printf("b) %d  %x\n", resultado, resultado);
  
  resultado= (x<<3)+(x<<2);
  printf("c) %d  %x\n", resultado, resultado);
}

int main() {

  int x;

  printf ("Indique o número a testar:\n");
  scanf("%d", &x);
  binoper(x);
  return 0;
}

